src/addon/NUOPC/src/NUOPC_Connector.F90

```diff
@@ -609,25 +609,22 @@ subroutine InitializeIPDv05p1(connector, importState, exportState, clock, rc)
       if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
         line=__LINE__, file=trim(name)//":"//FILENAME)) return  ! bail out
     elseif (trim(exportXferPolicy)=="transferAllAsNests") then
-      ! check name of provider component
-      call NUOPC_GetAttribute(importState, name="CompName", &
-        value=importProvider, rc=rc)
+      ! access importState namespace so it can be transferred to exportState
+      call NUOPC_GetAttribute(importState, name="Namespace", &
+        value=namespace, rc=rc)
       if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
         line=__LINE__, file=trim(name)//":"//FILENAME)) return  ! bail out
-      ! create nested state
-      exportNestedState = ESMF_StateCreate(name=trim(importProvider)//"-NestedState", rc=rc)
+      ! access name of exportState for nestedStateName construction for clarity
+      call ESMF_StateGet(exportState, name=stateName, rc=rc)
       if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
         line=__LINE__, file=trim(name)//":"//FILENAME)) return  ! bail out
-      ! set FieldTransferPolicy metadata for nested state
-      call NUOPC_SetAttribute(exportNestedState, "FieldTransferPolicy", "transferAll", rc=rc)
+      ! set namespace on exportState, creating a nestedState on acceptor VM
+      call NUOPC_AddNamespace(exportState, namespace=trim(namespace), &
+        nestedStateName=trim(stateName)//"-namespace:"//trim(namespace), &
```

Comment:
The nested state name is appearing in the NUOPC app prototype something like `atm-importstate-namespace:ocn` and this is aligned with the change in here.  So, this is little bit different than the convention that I was using before.  Are we plaining to go with this naming convention? If so, I need to couple of adjustment in my real application. Also, I think we need to document this convention in the documentation.

Reply:
Let's discuss this further next week (on a call, or under your PR). I am thinking your code should only be going off of the `Namespace` attribute, which is part of the NUOPC standard, and documented. The nested State name itself is a NUOPC implementation detail, and user code should not depend on it (I just made it something meaningful for the purpose of internal NUOPC development and debugging). But we can discuss this in more detail... I may be overlooking something.